lsp-dart.el

```diff
@@ -110,14 +110,27 @@ If unspecified, diagnostics will not be generated."
         (append (list (file-name-directory (buffer-file-name))) lsp-dart-extra-library-directories)
       lsp-dart-extra-library-directories)))
 
+(defun lsp-dart--dart-capabiliities (&optional sdk-root)
+  "Given an SDK-ROOT, return capabilities for that dart version.
+Default to `lsp-dart-get-sdk-dir' if no root is passed.
+Returns nil if an invalid SDK-ROOT has been passed."
+  (when-let* ((root (or sdk-root lsp-dart-sdk-dir (lsp-dart-get-sdk-dir)))
+              (dart-version (string-trim
+                             (with-temp-buffer
+                               (insert-file-contents (expand-file-name "version" root))
+                               (buffer-string)))))
+    `((canUseLanguageServer . ,(version<= "2.14.4" dart-version)))))
+
 (defun lsp-dart--server-command ()
   "Generate LSP startup command."
   (or lsp-dart-server-command
-      (list (lsp-dart-dart-command)
-            (expand-file-name (f-join (lsp-dart-get-sdk-dir) "bin/snapshots/analysis_server.dart.snapshot"))
-            "--lsp"
-            "--client-id emacs.lsp-dart"
-            (format "--client-version %s" lsp-dart-version-string))))
+      `(,(lsp-dart-dart-command)
+        ,@(if (alist-get 'canUseLanguageServer (lsp-dart--dart-capabiliities))
```

Comment:
Why can't we just use `(lsp-dart-version-at-least-p "2.14.4")` here?

Reply:
Rather than having opaque version checks everywhere I thought it made more sense to mirror what dart-code does and provide a function to list all the capabilities the current sdk provides. 

I also wasn't aware that function already existed, so I can make the capacities use that instead

Reply:
I see, I think it's ok to keep the capabilities but use the util function to get the version, WDYT?